src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/YFlowUpdateService.java

```diff
@@ -0,0 +1,319 @@
+        YFlowRequest updateRequest = null;
+        if (request.getYFlowId() != null) {
+            YFlow yFlow = yFlowRepository.findById(request.getYFlowId()).orElse(null);
+            updateRequest = YFlowRequestMapper.INSTANCE.toYFlowRequest(yFlow);
+        }
+
+        if (updateRequest == null) {
+            updateRequest = YFlowRequest.builder().build();
+        }
+
+        Optional.ofNullable(request.getYFlowId()).ifPresent(updateRequest::setYFlowId);
+        if (request.getSharedEndpoint() != null) {
+            if (updateRequest.getSharedEndpoint() == null) {
+                updateRequest.setSharedEndpoint(new FlowEndpoint(request.getSharedEndpoint().getSwitchId(),
+                        request.getSharedEndpoint().getPortNumber()));
+            } else {
+                SwitchId switchId = Optional.ofNullable(request.getSharedEndpoint().getSwitchId())
+                        .orElse(updateRequest.getSharedEndpoint().getSwitchId());
+                int portNumber = Optional.ofNullable(request.getSharedEndpoint().getPortNumber())
+                        .orElse(updateRequest.getSharedEndpoint().getPortNumber());
+                updateRequest.setSharedEndpoint(new FlowEndpoint(switchId, portNumber));
+            }
+        }
+        Optional.ofNullable(request.getMaximumBandwidth()).ifPresent(updateRequest::setMaximumBandwidth);
+        Optional.ofNullable(request.getPathComputationStrategy()).ifPresent(updateRequest::setPathComputationStrategy);
+        Optional.ofNullable(request.getEncapsulationType()).ifPresent(updateRequest::setEncapsulationType);
+        Optional.ofNullable(request.getMaxLatency()).ifPresent(updateRequest::setMaxLatency);
+        Optional.ofNullable(request.getMaxLatencyTier2()).ifPresent(updateRequest::setMaxLatencyTier2);
+        Optional.ofNullable(request.getIgnoreBandwidth()).ifPresent(updateRequest::setIgnoreBandwidth);
+        Optional.ofNullable(request.getPeriodicPings()).ifPresent(updateRequest::setPeriodicPings);
+        Optional.ofNullable(request.getPinned()).ifPresent(updateRequest::setPinned);
+        Optional.ofNullable(request.getPriority()).ifPresent(updateRequest::setPriority);
+        Optional.ofNullable(request.getStrictBandwidth()).ifPresent(updateRequest::setStrictBandwidth);
+        Optional.ofNullable(request.getDescription()).ifPresent(updateRequest::setDescription);
+        Optional.ofNullable(request.getAllocateProtectedPath()).ifPresent(updateRequest::setAllocateProtectedPath);
+
+        if (request.getSubFlows() != null && !request.getSubFlows().isEmpty()) {
+            if (updateRequest.getSubFlows() != null) {
+                for (SubFlowPatchDto subFlowPatch : request.getSubFlows()) {
+                    updateRequest.getSubFlows().stream()
+                            .filter(f -> f.getFlowId().equals(subFlowPatch.getFlowId()))
+                            .findAny()
+                            .ifPresent(subFlow -> {
```

Comment:
What should we do it there is no match?

Reply:
We create a request based on the received data for a partial update. If the data is incorrect, YFlowUpdateFsm will respond with an error.

Reply:
No. You are "truncating" data provided in request to the "subset" of sub-flows present in `target` at this moment. So the "check" inside "validation" action have no change to failed requrest in case of requests with "extra" sub-flow records.

Reply:
Yes, you are right. This has been reworked.